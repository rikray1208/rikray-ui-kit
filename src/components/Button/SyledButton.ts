
import {ButtonProps} from "./index";
import {Shape, Size, Variant} from "../types";
import styled, {css} from "styled-components";

type StyledProps = Omit<ButtonProps, 'onClick'>


function setAdditionalStyles(variant: Variant, isSuccess: boolean) {
    if( variant == 'Filled' ) {
        return css`
          background: ${({theme}) => isSuccess ? theme.Button.colorSuccess : theme.Button.colorError };

          &:hover {
            background: ${({theme}) => isSuccess ? theme.Button.colorHoverSuccess : theme.Button.colorHoverError};
          }
        `
    }

    if ( variant == 'Text' ) {
        return css`
          color: ${({theme}) => isSuccess ? theme.Button.colorSuccess : theme.Button.colorError };

          &:hover {
            color: ${({theme}) =>  isSuccess ? theme.Button.colorHoverSuccess : theme.Button.colorHoverError};
            background-color: ${({theme}) => isSuccess ? theme.Button.colorHoverSuccess + '10' : theme.Button.colorHoverError + '10'};
          }
        `
    }

    return css`
      border-color: ${({theme}) =>isSuccess ? theme.Button.colorSuccess : theme.Button.colorError};
      color: ${({theme}) => isSuccess ? theme.Button.colorSuccess : theme.Button.colorError};

      &:hover {
        background: ${({theme}) => isSuccess ? theme.Button.colorHoverSuccess + '06' : theme.Button.colorHoverError + '06'};
        border-color: ${({theme}) => isSuccess ? theme.Button.colorHoverSuccess : theme.Button.colorHoverError};
        color: ${({theme}) => isSuccess ? theme.Button.colorHoverSuccess : theme.Button.colorHoverError};
      }
    `
}

const Filled = css<StyledProps>`
  ${({theme, success, danger}) =>
          css`
            background: ${theme.Button.colorPrimary};
            color: ${theme.Button.colorWhite};

            &:hover {
              background: ${theme.Button.colorHoverBase};
            }
            &:disabled {
              background-color: ${theme.Button.colorDisable};
            }

            ${success && setAdditionalStyles('Filled', true)}
            ${danger && setAdditionalStyles('Filled', false)}

          `
  }
`;

const Outline = css<StyledProps>`
  ${({theme, success, danger}) => css`
    border-style: ${theme.Button.borderStyle};
    border-width: ${theme.Button.borderWidth};
    border-color: ${theme.Button.colorPrimary};
    color: ${theme.Button.colorPrimaryText};

    &:hover {
      border-color: ${theme.Button.colorHoverPrimary};
      color: ${theme.Button.colorHoverPrimary};
      background: ${theme.Button.colorHoverPrimary + '06 '};
    }
    &:disabled {
      border-color: ${theme.Button.colorDisable};
    }
    ${success && setAdditionalStyles('Outline', true)}
    ${danger && setAdditionalStyles('Outline', false)}
  `}
`;

const Dashed = css<StyledProps>`
  ${({theme, success, danger}) => css`
    border-style: ${theme.Button.buttonBorderDashed};
    border-width: ${theme.Button.borderWidth};
    border-color: ${theme.Button.colorPrimary};
    color: ${theme.Button.colorPrimaryText};

    &:hover {
      background: ${theme.Button.colorHoverBase + '15'};
    }
    &:disabled {
      border-color: ${theme.Button.colorDisable};
    }
    ${success && setAdditionalStyles('Dashed', true)}
    ${danger && setAdditionalStyles('Dashed', false)}
  `}
`;

const Text = css<StyledProps>`
  color: ${({theme}) => theme.Button.colorPrimaryText};

  &:hover {
    background: ${({theme}) => theme.Button.colorHoverPrimary + '15'};
  }
  &:disabled {
    background: none;
  }

  ${({success}) => success && setAdditionalStyles('Text', true)}
  ${({danger}) => danger && setAdditionalStyles('Text', false)}

`;

const variants = {
    Filled,
    Outline,
    Dashed,
    Text
}

function setSize(size: Size) {
    if (size == 'Small') {
        return css`
          font-size: ${({theme}) => theme.Button["fontSize-sm"] + 'rem'};
          line-height: ${({theme}) => theme.Button["lineHeight-sm"] + 'rem'};
        `
    }

    if (size == 'Medium') {
        return css`
          font-size: ${({theme}) => theme.Button["fontSize-base"] + 'rem'};
          line-height: ${({theme}) => theme.Button["lineHeight-base"] + 'rem'};
        `
    }

    if (size == 'Large') {
        return css`
          font-size: ${({theme}) => theme.Button["fontSize-lg"] + 'rem'};
          line-height: ${({theme}) => theme.Button["lineHeight-lg"] + 'rem'};
        `
    }
}

function setShape(shape: Shape) {
    if (shape == 'Rectangle') {
        return css`
          border-radius: ${({theme}) => theme.Button["borderRadius-xs"] + 'px'};
        `
    }

    if (shape == 'Round') {
        return css`
          border-radius: ${({theme}) => theme.Button["borderRadius-lg"] + 'px'};
        `
    }

    if (shape == 'SemiRound') {
        return css`
          border-radius: ${({theme}) => theme.Button["borderRadius-sm"] + 'px'};
        `
    }
}


const ButtonStyle = css<StyledProps>`
  ${({ theme, shape, size, variant }) => css`
    font-size: ${theme.Button["fontSize-base"] + 'rem'};
    line-height: ${theme.Button["lineHeight-base"] + 'rem'};
    font-family: ${theme.Button["fontFamily"]};

    background: transparent;
    border: none;

    color: ${theme.Button.colorTextBase};

    padding: ${theme.Button["padding-1"] + 'rem'} ${theme.Button["padding-3"] + 'rem'};

    transition: all ${theme.Button["durationBase"] + 'ms'} ${theme.Button['transitionEaseInOut']};

    cursor: pointer;

    ${
            shape && setShape(shape)
    }
    ${
            size && setSize(size)
    }

    &:not([disabled]):active {
      transform: scale(${theme.Button.scaleButtonActive});
    }

    ${variant && variants[variant]}

    &:disabled {
      color: ${theme.Button.colorDisableText};
      cursor: not-allowed;
      background-color: ${theme.Button.colorDisable}
    }



  `}
`

const StyledButton = styled.button<ButtonProps>`
  ${ButtonStyle}
`
export default StyledButton
