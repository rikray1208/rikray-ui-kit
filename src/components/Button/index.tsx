import React, {FC, PropsWithChildren} from 'react';
import styled from "styled-components";
import BaseTheme from "../../theme";

export type Variant = 'text' | 'contained' | 'outline';

interface ButtonProps {
    variant: Variant,
    disabled: boolean,
    onClick?: () => void,
    color: any,
    size: any,
}

interface StyledProps {
    color?: string,
    variant: Variant,
    size: string
}


const StyledButton = styled.button<StyledProps>`
  background-color: ${({color, variant}) => variant == 'text' || variant == 'outline' ? 'inherit' : color};
  padding: 16px;
  border: ${({variant, color}) => variant === 'outline' ? `1px solid ${color}` : 'none'};
  border-radius: ${({variant}) => variant !== 'text' ? '16px' : '0'};
  color: ${({variant, color}) => variant == 'text' || variant == 'outline' ? color : "white"};
  font-weight: bold;
  font-size: ${({size}) => size};
  
`

const Button: FC<PropsWithChildren<ButtonProps>> = ({children, ...props}) => {
    return (
        <div>
            <StyledButton {...props}>
                {children}
            </StyledButton>
        </div>
    );
};

export default Button;