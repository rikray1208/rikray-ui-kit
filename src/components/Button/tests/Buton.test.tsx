import React from "react";
import Button from "../index";

import renderer from "react-test-renderer";

import 'jest-styled-components'
import {ThemeProvider} from "../../../index";

const assertReactElement = (element: React.ReactElement) => {
  const component = renderer.create(element);
  return component.toJSON();
};

describe('button', () => {
    it('should return a Button', () =>{
        const element = (
            <ThemeProvider><Button></Button></ThemeProvider>
        );
        expect(assertReactElement(element)).toMatchSnapshot();
    })
    it('should return a Button with children', () =>{
        const element = (
            <ThemeProvider><Button>Button</Button></ThemeProvider>
        );
        expect(assertReactElement(element)).toMatchSnapshot();
    })
    it('should return a Button with disabled', () =>{
        const element = (
            <ThemeProvider><Button disabled={true}>Button</Button></ThemeProvider>
        );
        expect(assertReactElement(element)).toMatchSnapshot();
    })
    it('should return a Outline Button', () =>{
        const element = (
            <ThemeProvider><Button variant={"Outline"}>Button</Button></ThemeProvider>
        );
        expect(assertReactElement(element)).toMatchSnapshot();
    })
    it('should return a Dashed Button', () =>{
        const element = (
            <ThemeProvider><Button variant={"Dashed"}>Button</Button></ThemeProvider>
        );
        expect(assertReactElement(element)).toMatchSnapshot();
    })
    it('should return a Text Button', () =>{
        const element = (
            <ThemeProvider><Button variant={"Text"}>Button</Button></ThemeProvider>
        );
        expect(assertReactElement(element)).toMatchSnapshot();
    })
    it('should return a Large Button', () =>{
        const element = (
            <ThemeProvider><Button size={"Large"}>Button</Button></ThemeProvider>
        );
        expect(assertReactElement(element)).toMatchSnapshot();
    })
    it('should return a Medium Button', () =>{
        const element = (
            <ThemeProvider><Button size={"Medium"}>Button</Button></ThemeProvider>
        );
        expect(assertReactElement(element)).toMatchSnapshot();
    })
    it('should return a Small Button', () =>{
        const element = (
            <ThemeProvider><Button size={"Small"}>Button</Button></ThemeProvider>
        );
        expect(assertReactElement(element)).toMatchSnapshot();
    })
    it('should return a Rectangle Button', () =>{
        const element = (
            <ThemeProvider><Button shape={"Rectangle"}>Button</Button></ThemeProvider>
        );
        expect(assertReactElement(element)).toMatchSnapshot();
    })
    it('should return a Round Button', () =>{
        const element = (
            <ThemeProvider><Button shape={"Round"}>Button</Button></ThemeProvider>
        );
        expect(assertReactElement(element)).toMatchSnapshot();
    })
    it('should return a SemiRound Button', () =>{
        const element = (
            <ThemeProvider><Button shape={"SemiRound"}>Button</Button></ThemeProvider>
        );
        expect(assertReactElement(element)).toMatchSnapshot();
    })
    it('should return a success Button', () =>{
        const element = (
            <ThemeProvider><Button success={true}>Button</Button></ThemeProvider>
        );
        expect(assertReactElement(element)).toMatchSnapshot();
    })
    it('should return a danger Button', () =>{
        const element = (
            <ThemeProvider><Button danger={true}>Button</Button></ThemeProvider>
        );
        expect(assertReactElement(element)).toMatchSnapshot();
    })
})


