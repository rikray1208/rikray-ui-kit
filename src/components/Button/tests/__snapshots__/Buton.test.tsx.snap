// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button should return a Button 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,'Helvetica Neue',Arial,'Noto Sans',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol','Noto Color Emoji';
  background: transparent;
  border: none;
  color: #000000;
  padding: 0.30000000000000004rem 0.8rem;
  -webkit-transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  cursor: pointer;
  background: #3B74F3;
  color: #ffffff;
  border-radius: 8px;
  font-size: 1rem;
  line-height: 1.5rem;
}

.c0:not([disabled]):active {
  -webkit-transform: scale(0.92);
  -ms-transform: scale(0.92);
  transform: scale(0.92);
}

.c0:hover {
  background: #437cfb;
}

.c0:disabled {
  background-color: rgba(0,0,0,0.12);
}

.c0:disabled {
  color: rgba(0,0,0,0.25);
  cursor: not-allowed;
  background-color: rgba(0,0,0,0.12);
}

<button
  className="c0"
  disabled={false}
  shape="SemiRound"
  size="Medium"
/>
`;

exports[`button should return a Button with children 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,'Helvetica Neue',Arial,'Noto Sans',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol','Noto Color Emoji';
  background: transparent;
  border: none;
  color: #000000;
  padding: 0.30000000000000004rem 0.8rem;
  -webkit-transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  cursor: pointer;
  background: #3B74F3;
  color: #ffffff;
  border-radius: 8px;
  font-size: 1rem;
  line-height: 1.5rem;
}

.c0:not([disabled]):active {
  -webkit-transform: scale(0.92);
  -ms-transform: scale(0.92);
  transform: scale(0.92);
}

.c0:hover {
  background: #437cfb;
}

.c0:disabled {
  background-color: rgba(0,0,0,0.12);
}

.c0:disabled {
  color: rgba(0,0,0,0.25);
  cursor: not-allowed;
  background-color: rgba(0,0,0,0.12);
}

<button
  className="c0"
  disabled={false}
  shape="SemiRound"
  size="Medium"
>
  Button
</button>
`;

exports[`button should return a Button with disabled 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,'Helvetica Neue',Arial,'Noto Sans',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol','Noto Color Emoji';
  background: transparent;
  border: none;
  color: #000000;
  padding: 0.30000000000000004rem 0.8rem;
  -webkit-transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  cursor: pointer;
  background: #3B74F3;
  color: #ffffff;
  border-radius: 8px;
  font-size: 1rem;
  line-height: 1.5rem;
}

.c0:not([disabled]):active {
  -webkit-transform: scale(0.92);
  -ms-transform: scale(0.92);
  transform: scale(0.92);
}

.c0:hover {
  background: #437cfb;
}

.c0:disabled {
  background-color: rgba(0,0,0,0.12);
}

.c0:disabled {
  color: rgba(0,0,0,0.25);
  cursor: not-allowed;
  background-color: rgba(0,0,0,0.12);
}

<button
  className="c0"
  disabled={true}
  shape="SemiRound"
  size="Medium"
>
  Button
</button>
`;

exports[`button should return a Dashed Button 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,'Helvetica Neue',Arial,'Noto Sans',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol','Noto Color Emoji';
  background: transparent;
  border: none;
  color: #000000;
  padding: 0.30000000000000004rem 0.8rem;
  -webkit-transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  cursor: pointer;
  border-style: dashed;
  border-width: 1px;
  border-color: #3B74F3;
  color: #3B74F3;
  border-radius: 8px;
  font-size: 1rem;
  line-height: 1.5rem;
}

.c0:not([disabled]):active {
  -webkit-transform: scale(0.92);
  -ms-transform: scale(0.92);
  transform: scale(0.92);
}

.c0:hover {
  background: #f5f5f515;
}

.c0:disabled {
  border-color: rgba(0,0,0,0.12);
}

.c0:disabled {
  color: rgba(0,0,0,0.25);
  cursor: not-allowed;
  background-color: rgba(0,0,0,0.12);
}

<button
  className="c0"
  disabled={false}
  shape="SemiRound"
  size="Medium"
>
  Button
</button>
`;

exports[`button should return a Large Button 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,'Helvetica Neue',Arial,'Noto Sans',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol','Noto Color Emoji';
  background: transparent;
  border: none;
  color: #000000;
  padding: 0.30000000000000004rem 0.8rem;
  -webkit-transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  cursor: pointer;
  background: #3B74F3;
  color: #ffffff;
  border-radius: 8px;
  font-size: 1.125rem;
  line-height: 1.75rem;
}

.c0:not([disabled]):active {
  -webkit-transform: scale(0.92);
  -ms-transform: scale(0.92);
  transform: scale(0.92);
}

.c0:hover {
  background: #437cfb;
}

.c0:disabled {
  background-color: rgba(0,0,0,0.12);
}

.c0:disabled {
  color: rgba(0,0,0,0.25);
  cursor: not-allowed;
  background-color: rgba(0,0,0,0.12);
}

<button
  className="c0"
  disabled={false}
  shape="SemiRound"
  size="Large"
>
  Button
</button>
`;

exports[`button should return a Medium Button 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,'Helvetica Neue',Arial,'Noto Sans',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol','Noto Color Emoji';
  background: transparent;
  border: none;
  color: #000000;
  padding: 0.30000000000000004rem 0.8rem;
  -webkit-transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  cursor: pointer;
  background: #3B74F3;
  color: #ffffff;
  border-radius: 8px;
  font-size: 1rem;
  line-height: 1.5rem;
}

.c0:not([disabled]):active {
  -webkit-transform: scale(0.92);
  -ms-transform: scale(0.92);
  transform: scale(0.92);
}

.c0:hover {
  background: #437cfb;
}

.c0:disabled {
  background-color: rgba(0,0,0,0.12);
}

.c0:disabled {
  color: rgba(0,0,0,0.25);
  cursor: not-allowed;
  background-color: rgba(0,0,0,0.12);
}

<button
  className="c0"
  disabled={false}
  shape="SemiRound"
  size="Medium"
>
  Button
</button>
`;

exports[`button should return a Outline Button 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,'Helvetica Neue',Arial,'Noto Sans',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol','Noto Color Emoji';
  background: transparent;
  border: none;
  color: #000000;
  padding: 0.30000000000000004rem 0.8rem;
  -webkit-transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  cursor: pointer;
  border-style: solid;
  border-width: 1px;
  border-color: #3B74F3;
  color: #3B74F3;
  border-radius: 8px;
  font-size: 1rem;
  line-height: 1.5rem;
}

.c0:not([disabled]):active {
  -webkit-transform: scale(0.92);
  -ms-transform: scale(0.92);
  transform: scale(0.92);
}

.c0:hover {
  border-color: #437cfb;
  color: #437cfb;
  background: #437cfb06;
}

.c0:disabled {
  border-color: rgba(0,0,0,0.12);
}

.c0:disabled {
  color: rgba(0,0,0,0.25);
  cursor: not-allowed;
  background-color: rgba(0,0,0,0.12);
}

<button
  className="c0"
  disabled={false}
  shape="SemiRound"
  size="Medium"
>
  Button
</button>
`;

exports[`button should return a Rectangle Button 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,'Helvetica Neue',Arial,'Noto Sans',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol','Noto Color Emoji';
  background: transparent;
  border: none;
  color: #000000;
  padding: 0.30000000000000004rem 0.8rem;
  -webkit-transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  cursor: pointer;
  background: #3B74F3;
  color: #ffffff;
  border-radius: 4px;
  font-size: 1rem;
  line-height: 1.5rem;
}

.c0:not([disabled]):active {
  -webkit-transform: scale(0.92);
  -ms-transform: scale(0.92);
  transform: scale(0.92);
}

.c0:hover {
  background: #437cfb;
}

.c0:disabled {
  background-color: rgba(0,0,0,0.12);
}

.c0:disabled {
  color: rgba(0,0,0,0.25);
  cursor: not-allowed;
  background-color: rgba(0,0,0,0.12);
}

<button
  className="c0"
  disabled={false}
  shape="Rectangle"
  size="Medium"
>
  Button
</button>
`;

exports[`button should return a Round Button 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,'Helvetica Neue',Arial,'Noto Sans',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol','Noto Color Emoji';
  background: transparent;
  border: none;
  color: #000000;
  padding: 0.30000000000000004rem 0.8rem;
  -webkit-transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  cursor: pointer;
  background: #3B74F3;
  color: #ffffff;
  border-radius: 12px;
  font-size: 1rem;
  line-height: 1.5rem;
}

.c0:not([disabled]):active {
  -webkit-transform: scale(0.92);
  -ms-transform: scale(0.92);
  transform: scale(0.92);
}

.c0:hover {
  background: #437cfb;
}

.c0:disabled {
  background-color: rgba(0,0,0,0.12);
}

.c0:disabled {
  color: rgba(0,0,0,0.25);
  cursor: not-allowed;
  background-color: rgba(0,0,0,0.12);
}

<button
  className="c0"
  disabled={false}
  shape="Round"
  size="Medium"
>
  Button
</button>
`;

exports[`button should return a SemiRound Button 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,'Helvetica Neue',Arial,'Noto Sans',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol','Noto Color Emoji';
  background: transparent;
  border: none;
  color: #000000;
  padding: 0.30000000000000004rem 0.8rem;
  -webkit-transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  cursor: pointer;
  background: #3B74F3;
  color: #ffffff;
  border-radius: 8px;
  font-size: 1rem;
  line-height: 1.5rem;
}

.c0:not([disabled]):active {
  -webkit-transform: scale(0.92);
  -ms-transform: scale(0.92);
  transform: scale(0.92);
}

.c0:hover {
  background: #437cfb;
}

.c0:disabled {
  background-color: rgba(0,0,0,0.12);
}

.c0:disabled {
  color: rgba(0,0,0,0.25);
  cursor: not-allowed;
  background-color: rgba(0,0,0,0.12);
}

<button
  className="c0"
  disabled={false}
  shape="SemiRound"
  size="Medium"
>
  Button
</button>
`;

exports[`button should return a Small Button 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,'Helvetica Neue',Arial,'Noto Sans',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol','Noto Color Emoji';
  background: transparent;
  border: none;
  color: #000000;
  padding: 0.30000000000000004rem 0.8rem;
  -webkit-transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  cursor: pointer;
  background: #3B74F3;
  color: #ffffff;
  border-radius: 8px;
  font-size: 0.875rem;
  line-height: 1.25rem;
}

.c0:not([disabled]):active {
  -webkit-transform: scale(0.92);
  -ms-transform: scale(0.92);
  transform: scale(0.92);
}

.c0:hover {
  background: #437cfb;
}

.c0:disabled {
  background-color: rgba(0,0,0,0.12);
}

.c0:disabled {
  color: rgba(0,0,0,0.25);
  cursor: not-allowed;
  background-color: rgba(0,0,0,0.12);
}

<button
  className="c0"
  disabled={false}
  shape="SemiRound"
  size="Small"
>
  Button
</button>
`;

exports[`button should return a Text Button 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,'Helvetica Neue',Arial,'Noto Sans',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol','Noto Color Emoji';
  background: transparent;
  border: none;
  color: #000000;
  padding: 0.30000000000000004rem 0.8rem;
  -webkit-transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  cursor: pointer;
  color: #3B74F3;
  border-radius: 8px;
  font-size: 1rem;
  line-height: 1.5rem;
}

.c0:not([disabled]):active {
  -webkit-transform: scale(0.92);
  -ms-transform: scale(0.92);
  transform: scale(0.92);
}

.c0:hover {
  background: #437cfb15;
}

.c0:disabled {
  background: none;
}

.c0:disabled {
  color: rgba(0,0,0,0.25);
  cursor: not-allowed;
  background-color: rgba(0,0,0,0.12);
}

<button
  className="c0"
  disabled={false}
  shape="SemiRound"
  size="Medium"
>
  Button
</button>
`;

exports[`button should return a danger Button 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,'Helvetica Neue',Arial,'Noto Sans',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol','Noto Color Emoji';
  background: transparent;
  border: none;
  color: #000000;
  padding: 0.30000000000000004rem 0.8rem;
  -webkit-transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  cursor: pointer;
  background: #3B74F3;
  color: #ffffff;
  background: #ff0e0e;
  border-radius: 8px;
  font-size: 1rem;
  line-height: 1.5rem;
}

.c0:not([disabled]):active {
  -webkit-transform: scale(0.92);
  -ms-transform: scale(0.92);
  transform: scale(0.92);
}

.c0:hover {
  background: #437cfb;
}

.c0:disabled {
  background-color: rgba(0,0,0,0.12);
}

.c0:hover {
  background: #ff1616;
}

.c0:disabled {
  color: rgba(0,0,0,0.25);
  cursor: not-allowed;
  background-color: rgba(0,0,0,0.12);
}

<button
  className="c0"
  disabled={false}
  shape="SemiRound"
  size="Medium"
>
  Button
</button>
`;

exports[`button should return a success Button 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,'Helvetica Neue',Arial,'Noto Sans',sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol','Noto Color Emoji';
  background: transparent;
  border: none;
  color: #000000;
  padding: 0.30000000000000004rem 0.8rem;
  -webkit-transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  transition: all 200ms cubic-bezier(0.4,0,0.2,1);
  cursor: pointer;
  background: #3B74F3;
  color: #ffffff;
  background: #00AB66;
  border-radius: 8px;
  font-size: 1rem;
  line-height: 1.5rem;
}

.c0:not([disabled]):active {
  -webkit-transform: scale(0.92);
  -ms-transform: scale(0.92);
  transform: scale(0.92);
}

.c0:hover {
  background: #437cfb;
}

.c0:disabled {
  background-color: rgba(0,0,0,0.12);
}

.c0:hover {
  background: #8b36e;
}

.c0:disabled {
  color: rgba(0,0,0,0.25);
  cursor: not-allowed;
  background-color: rgba(0,0,0,0.12);
}

<button
  className="c0"
  disabled={false}
  shape="SemiRound"
  size="Medium"
>
  Button
</button>
`;
